types/next/error.d.ts

```diff
@@ -10,5 +10,5 @@ export interface DefaultErrorIProps {
 }
 
 export default class Error<P = {}> extends React.Component<P & DefaultErrorIProps> {
-    static getInitialProps(context: NextContext): DefaultErrorIProps;
+    static getInitialProps(context: NextContext): Promise<DefaultErrorIProps>;
```

Comment:
Can we change it to `DefaultErrorIProps | Promise<DefaultErrorIProps>`? The method in the exported Error class from Next.js returns an object (https://github.com/zeit/next.js/blob/canary/packages/next/pages/_error.js#L7)

Reply:
Well, this is actually interesting, I though it was a mistake from the typings, but it looks like a problem with my implementation, I used the `async` keyword on `getInitialProps` when I actually did not need it.

Do you think the proposal of having it as potentially a Promise is the good solution? Or should it always be NOT a Promise?

If the latter, then I guess we can close this PR.

Reply:
It's a weird one that happened several times in the past as well - and I still didn't completely wrap my head around it. The typings match the exported classes. When you extend classes the methods should keep the same signatures. But Next.js multiple return types.

We can keep the PR with my suggested change. Though we should think about it properly and holistically - this happens to all Next.js components (App, Document, Error).